src/pages/Vault/index.tsx

```diff
@@ -273,7 +271,8 @@ const Vault = (): JSX.Element => {
             ))}
           </div>
         </div>
-        {interactionEnabled && (
+        {/* Check interaction with the vault */}
+        {vaultClientLoaded && address === selectedVaultAddress && (
```

Comment:
@tomjeatt 
I found an alternative way we could compromise. Does it look good to you?

Reply:
@tomjeatt 
Could you please approve this PR as soon as possible? I'm keeping resolving conflicts due to merged PRs.

Reply:
Not really but let's not delay this any longer :) we can talk about it tomorrow

Reply:
Thank you for approving.